cargo-psp/src/main.rs

```diff
@@ -205,126 +205,100 @@ fn main() {
-                    ("-d", "REGION", config.region.as_ref().map(u32::to_string)),
-                    ("-s", "TITLE_0", config.title_jp.clone()),
-                    ("-s", "TITLE_2", config.title_fr.clone()),
-                    ("-s", "TITLE_3", config.title_es.clone()),
-                    ("-s", "TITLE_4", config.title_de.clone()),
-                    ("-s", "TITLE_5", config.title_it.clone()),
-                    ("-s", "TITLE_6", config.title_nl.clone()),
-                    ("-s", "TITLE_7", config.title_pt.clone()),
-                    ("-s", "TITLE_8", config.title_ru.clone()),
-                    ("-s", "UPDATER_VER", config.updater_version.clone()),
-                ];
-
-                Command::new("mksfo")
-                    // Add the optional config args
-                    .args({
-                        config_args
-                            .into_iter()
-                            // Filter through all the values that are not `None`
-                            .filter_map(|(f, k, v)| v.map(|v| (f, k, v)))
-                            // Map into 2 arguments, e.g. "-s" "NAME=VALUE"
-                            .flat_map(|(flag, key, value)| {
-                                vec![flag.into(), format!("{}={}", key, value)]
-                            })
-                    })
-                    .arg(config.title.clone().unwrap_or(target.name))
-                    .arg(&sfo_path)
-                    .stdin(Stdio::inherit())
-                    .stdout(Stdio::inherit())
-                    .stderr(Stdio::inherit())
-                    .output()
-                    .expect("failed to run mksfo");
-
-                Command::new("pack-pbp")
-                    .arg(&pbp_path)
-                    .arg(&sfo_path)
-                    .arg(config.xmb_icon_png.clone().unwrap_or("NULL".into()))
-                    .arg(config.xmb_icon_pmf.clone().unwrap_or("NULL".into()))
-                    .arg(
-                        config
-                            .xmb_background_overlay_png
-                            .clone()
-                            .unwrap_or("NULL".into()),
-                    )
-                    .arg(config.xmb_background_png.clone().unwrap_or("NULL".into()))
-                    .arg(config.xmb_music_at3.clone().unwrap_or("NULL".into()))
-                    .arg(&prx_path)
-                    .arg(config.psar.clone().unwrap_or("NULL".into()))
-                    .stdin(Stdio::inherit())
-                    .stdout(Stdio::inherit())
-                    .stderr(Stdio::inherit())
-                    .output()
-                    .expect("failed to run pack-pbp");
-            }
-        }
+    // TODO: Error if no bin is ever found.
+    for elf_path in executables {
+        let prx_path = elf_path.with_extension("prx");
+
+        let sfo_path = elf_path.with_extension("param.sfo");
+        let pbp_path = elf_path.with_extension("eboot.pbp");
```

Comment:
Let's keep the EBOOT name as `EBOOT.PBP`, and the SFO name as `PARAM.SFO`. As it stands right now, this would require explaining in the README that the `crate_name.eboot.pbp` needs to be renamed to `EBOOT.PBP`. While this does allow multiple PSP executables in a single workspace, it adds friction to all users, in return for a use case that is likely so rare nobody will ever use it.

Reply:
Or perhaps it is possible to only do this if multiple binaries are detected?

Reply:
What about something like
/target/mipsel-sony-psp/release/
------------------------------------------bin1/
------------------------------------------------/EBOOT.PBP
------------------------------------------------/bin1.prx
-----------------------------------------------/PARAM.SFO
----------------------------------------bin2/
----------------------------------------------/EBOOT.PBP
----------------------------------------------/bin2.prx
---------------------------------------------/PARAM.SFO

where bin1 and bin2 are the "elf_path"s in the workspace

Reply:
I would add that vitasdk toolchain for psvita (which is based on pspsdk) uses velf, param-sfo, self and vpk extensions for cmake, while pspsdk assumes there's no such thing as multi executable projects. Vitasdk made this jump so i ask you to do that too.

Reply:
I agree with sofa that what you're proposing adds unnecessary overhead of renaming files for end-users.

Reply:
Then make a descision.

Reply:
@ZetaNumbers Let's take things down a notch, there is no need to make snarky comments like that. I have already made a decision in my first comment on this PR: **Let's keep `EBOOT.PBP` for single binary projects.** Having it conditionally rename to `elf_name.eboot.pbp` is fine if there are multiple binaries, I have no preference there.

I understand you are making a purity argument, but you should remember that for a project with incredibly high friction for new users (new hardware, new graphics libraries, new app deployment), having to rename `EBOOT.PBP` every time an application is copied to the memory stick is annoying enough to kill continued iteration. Nobody is writing PSP apps for a living, so anything that makes the process unnecessarily annoying is not useful.

Reply:
Also, while we're trying to cool things off, I want you to know that your contributions are valued, but we need to work together to decide what is best for our users.

Reply:
What about when switching from a single binary to a workspace and vice versa? I think it should delete the old `EBOOT.BIN`s then. I think we should have uppercase extensions too then, like `hello-world.EBOOT.BIN`.

Also sorry if i was looking snarky there. I didn't meant to, i just don't want to run ahead of the train.

Reply:
> What about when switching from a single binary to a workspace and vice versa?

That's a good catch, but ultimately I don't think it's a usecase worth adding code for. If the user wants to clean up the stale files when they switch back and forth (which will be rare), they can do a `cargo clean`

Reply:
> > What about when switching from a single binary to a workspace and vice versa?
> 
> That's a good catch, but ultimately I don't think it's a usecase worth adding code for. If the user wants to clean up the stale files when they switch back and forth (which will be rare), they can do a `cargo clean` 

It's not about stale files, but to break the user's setup, if it would ever rely upon outdated files. I remember these "relying upon outdated files" errors to be especially annoying.

Reply:
Sure, with that said deleting the old EBOOT makes sense in a workspace if you want to implement it.

Reply:
Also there're multi-bin packages, with main.rs and bin/other.rs. I guess the main binary should remain prefixless, but not the others of course.

> Sure, with that said deleting the old EBOOT makes sense in a workspace if you want to implement it.

I wouldn't say so for the user. But it's easy to implement so, it shouldn't matter?

Reply:
It doesn't matter to me, you can decide. As long as the original behaviour of `EBOOT.PBP` is preserved for non-workspace projects this PR LGTM.

Reply:
Implemented in 8e492398967bf3a7767f325352a90a5cbcb577c6

Reply:
Actually this requires some more work to do. Right now it reduces file name on any single executable build, even in the workspace.

Reply:
Done